src/modules/sessionize/hooks.ts

```diff
@@ -1,26 +1,129 @@
+  const [parsedResult, setParsedResult] = useState<SessionizeViewAllSchemaType | undefined>(undefined)
+  const { data, error, isLoading } = useSWR(
     `https://sessionize.com/api/v2/${process.env.NEXT_PUBLIC_SESSIONIZE_ID}/view/All`,
     fetcher
   )
 
-  if (error) {
-    console.error(error)
-    return null
-  }
-  if (!data) {
-    return null
-  }
+  useEffect(() => {
+    if (!data) return
 
-  const parsedResult = sessionizeViewAllSchema.safeParse(data)
-  if (!parsedResult.success) {
-    console.error(parsedResult.error)
-    return null
-  }
+    const parsed = sessionizeViewAllSchema.safeParse(data)
+    if (!parsed.success) {
+      console.error(parsed.error)
+    } else {
+      setParsedResult(parsed.data)
+    }
+  }, [data])
+
+  const sessionTypes = useMemo(() => {
+    const category = parsedResult?.categories.find(c => c.id === categorySessionType)
+    return category?.items ?? []
+  }, [parsedResult?.categories])
 
-  return parsedResult.data
+  const sessionLevels = useMemo(() => {
+    const category = parsedResult?.categories.find(c => c.id === categorySessionLevel)
+    return category?.items ?? []
+  }, [parsedResult?.categories])
+
+  /**
+   * Get the name of the room with the given ID.
+   */
+  const getRoomName: ReturnType['getRoomName'] = useCallback(
+    roomId => {
+      const roomName = parsedResult?.rooms.find(room => room.id === roomId)?.name
+      if (!roomName) {
+        throw new Error(`Invalid roomId: ${roomId}`)
+      }
+      return roomName
+    },
+    [parsedResult?.rooms]
+  )
+
+  /**
+   * Get the session ID from the given session.
+   */
+  const getSessionNumber: ReturnType['getSessionNumber'] = useCallback(session => {
+    const sessionNum = session.questionAnswers.find(q => q.questionId === questionSessionNumber)
+    if (!sessionNum) {
+      throw new Error(`Invalid session: ${session}`)
+    }
+    return sessionNum.answerValue
+  }, [])
+
+  /**
+   * Get the speaker info from the given session.
+   */
+  const getSpeaker: ReturnType['getSpeaker'] = useCallback(
+    speakerId => {
+      const speaker = parsedResult?.speakers.find(s => s.id === speakerId)
+      if (!speaker) {
+        throw new Error(`Invalid speakerId: ${speakerId}`)
+      }
+      return speaker
+    },
+    [parsedResult?.speakers]
+  )
+
+  /**
+   * Get the session type from the given session.
+   */
+  const getSessionType: ReturnType['getSessionType'] = useCallback(
+    session => {
+      const sessionTypeId = session.categoryItems[0]
+      const sessionType = sessionTypes.find(c => c.id === sessionTypeId)
+      return sessionType?.name ?? ''
+    },
+    [sessionTypes]
+  )
+
+  /**
+   * Get the session level from the given session.
+   */
+  const getSessionLevel: ReturnType['getSessionLevel'] = useCallback(
+    session => {
+      const sessionLevelId = session.categoryItems[1]
+      const level = sessionLevels.find(c => c.id === sessionLevelId)
+      return level?.name ?? ''
```

Comment:
level が設定されていないセッションが多く、表示が不自然になっているのでデフォルト値を返してあげると良さそうです (デフォルトが all でいいのかは不明です)

```suggestion
      return level?.name ?? 'all'
```

![image](https://github.com/GoCon/2023/assets/40013676/cc3d8264-5333-4890-b9d7-a6c1612d9ef4)

![image](https://github.com/GoCon/2023/assets/40013676/9e7ff9af-905a-46ed-8542-43d93cd7b9fe)

Reply:
@taigakiyokawa 
入力していない箇所に `all` と出すのもちょっと気持ち悪いので、入力されていない場合は `Level:` の項目自体を非表示とするようにしてみました。
いかがでしょうか？

![image](https://github.com/GoCon/2023/assets/19329/ae0aefcb-79b9-41b2-9a03-b6bc78fc1beb)

Reply:
めっちゃ良いと思います！ 💯